azure/functions_worker/bindings/queue.py

```diff
@@ -60,6 +60,10 @@ def check_input_type_annotation(
             cls, pytype: type, datatype: protos.BindingInfo.DataType) -> bool:
         if datatype is protos.BindingInfo.undefined:
             return issubclass(pytype, azf_abc.QueueMessage)
+        elif (datatype is protos.BindingInfo.binary):
+            return issubclass(pytype, bytes)
```

Comment:
what about other bindings? shouldn't they be updated to handle dataType binary and string? what about outputbindings?

Reply:
Output bindings already support elementary types, where applicable.

> what about other bindings?

Which bindings do you have in mind?  Is there actually value in providing raw input for all bindings given that rich bindings already exist?  I'm thinking an abundance of options here would create confusion without actual benefit.

Reply:
Goal of this work item is to enable all bindings to work when `datatype` is specified in `function.json`
This means that, worker when issue #135 is addressed, users should be able to use functions without adding reference to azure.functions

Reply:
This still seems to me like a step backwards.  I understand why raw interface might be useful for when there is no canonical `azure.functions` binding yet, essentially as a workaround.

But our original direction with the design of the worker was to expose well-documented Pythonic interfaces as the API, not raw, untyped, data.  Promoting the latter creates a strong binding to the internals, and that may restrict our ability to upgrade existing users to newer versions of the platform.

Reply:
Using azure.functions is still the recommended approach. All the documentation and the samples will be using types in azure.functions package. 

Again, this work item is to make python worker an interface layer that converts to requested types.